Model: Update view after message edit

mkp6781 opened the pull request:
This PR attempts to maintain UI design after message edit by user.
This has been achieved by handling the `hanlde_update_message` event in a better manner and updating UI by creating message box when required.

Fixes #833

mkp6781 committed:
Model: Update view after message edit

This PR attempts to maintain UI design after message edit by user.
This has been achieved by handling the update_message event in a better manner and updating UI by creating message box when required.

Ezio-Sarthak commented:
@mkp6781 Thanks for working on this issue! It seems that you'll need to update the commit title and text as per Zulip's [commit message guidelines](http://zulip.readthedocs.io/en/latest/contributing/version-control.html) :)

For more clarification, do refer to [Development tasks](https://github.com/zulip/zulip-terminal#development-tasks) :+1:

zee-bit commented on zulipterminal/model.py:
```diff
@@ -1123,7 +1123,10 @@ def _update_rendered_view(self, msg_id: int) -> None:
                         and msg_box.message['subject'] != self.narrow[1][1]):
+                    msg_pos = view.message_view.log.index(msg_w)
                     view.message_view.log.remove(msg_w)
+                    prev_msg_sender = msg_box.last_message['sender_email']
```
The tests seem to be failing on this line. You need to specify a mock value of `sender_email` in the `test_model.py` module.

zee-bit reviewed (changes requested):
@mkp6781 You have some failed tests in your PR, that you need to fix. Also as pointed out by @Ezio-Sarthak your commit title needs to be updated too, but that is not the reason for this failing test.

Also make sure that before submitting a PR, you run `make check`(or the other command depending on how you installed ZT, see [here](https://github.com/zulip/zulip-terminal#development-tasks)), so that you can fix any linting errors or failed tests locally.

mkp6781 committed:
model: Shows elided date/sender fields after topic edit

Previously removal of earlier messages also removed the date/sender
fields of lower messages in some cases.
These cases that occur during topic edit by a user are now handled
to display the elided date/sender fields.

Tests updated
Fixes #833
